test/cql-pytest/cassandra_tests/validation/entities/json_test.py

```diff
@@ -258,10 +258,12 @@ def testFromJsonFct(cql, test_keyspace):
             assert_rows(execute(cql, table, "SELECT k, booleanval FROM %s WHERE k = ?", 0), [0, False])
 
             # strings are also accepted
-            # Reproduces issue #7915
             execute(cql, table, "INSERT INTO %s (k, booleanval) VALUES (?, fromJson(?))", 0, "\"false\"")
             assert_rows(execute(cql, table, "SELECT k, booleanval FROM %s WHERE k = ?", 0), [0, False])
 
+            execute(cql, table, "INSERT INTO %s (k, booleanval) VALUES (?, fromJson(?))", 0, "\"   tRuE   \"")
+            assert_rows(execute(cql, table, "SELECT k, booleanval FROM %s WHERE k = ?", 0), [0, True])
+
```

Comment:
The tests in cassandra_tests/ are all translated directly from Cassandra's unit tests (I do these translations once in a while, the translation project isn't done yet...). I don't want us to "improve" on these tests and add more tests to them. Instead, I want us to add more tests in directly in cql-pytest/ - outside the cassandra_tests/ subdirectory.

You can see I already added a test/cql-pytest/test_json.py with quite a few tests that I wrote. A lot of them were based on specific bugs in the translated Cassandra test (e.g., `test_fromjson_boolean_string_*`) and if you have new ideas on things you want to test, you can add more tests there.

You can either add your case insensitivity check to the existing test `test_fromjson_boolean_string_prepared()`, or even create a new test function - note that one of the ideas in the cql-pytest framework is individual test function have very low overhead - not only do we not start Scylla for each individual test function, we don't even create a new table for each test function - they all share one table (table1 in test_json.py). So if you want you can add a new test for the case sensitivity separate from the existing test (in this case you can go either way).